### PR TITLE
Implement ACPI

### PR DESCRIPTION
**Pull request by AnErrupTion**
This PR implements the ACPI driver under ISA (not really ISA, but for simplicity's sake). I've also implemented a few changes, like in PIT and CMOS.
To make the ACPI driver possible though, I had to modify the HAL class to make some internal functions public.

### CODE REVIEW

File: Source/Mosa.DeviceDriver/ISA/PIT.cs [hunk 1]
```diff
@@ -104,5 +117,16 @@ public uint GetTickCount()
 		{
 			return tickCount;
 		}
+
+		/// <summary>
+		/// Waits for a specific time, in milliseconds.
+		/// </summary>
+		public void Wait(uint ms)
```
Comment (tgiphil) [inline Source/Mosa.DeviceDriver/ISA/PIT.cs:124]: I think this method should be external to PIT.

Comment (AnErrupTion) [reply Source/Mosa.DeviceDriver/ISA/PIT.cs:124]: Where should it be then? It uses properties of PIT so I think it should be in PIT.

File: Source/Mosa.DeviceSystem/HAL.cs [hunk 2]
```diff
@@ -54,7 +54,7 @@ public static void ProcessInterrupt(byte irq)
 		/// </summary>
 		/// <param name="port">The port number.</param>
 		/// <returns></returns>
-		internal static BaseIOPortReadWrite GetReadWriteIOPort(ushort port)
+		public static BaseIOPortReadWrite GetReadWriteIOPort(ushort port)
```
Comment (tgiphil) [inline Source/Mosa.DeviceSystem/HAL.cs:57]: Not sure about this. It breaks the abstraction. But I understand the dilemma here.

Comment (AnErrupTion) [reply Source/Mosa.DeviceSystem/HAL.cs:57]: It compiles fine and runs fine, so I guess we can keep it (I mean I don't really have a choice either way :/)

File: Source/Mosa.DeviceDriver/ISA/ACPI.cs [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+				rsdpPtr = (Pointer)addr;
+
+				if (rsdpPtr.Load64() == 0x2052545020445352) // 'RSD PTR '
+					Descriptor = (RSDPDescriptor*)rsdpPtr;
+			}
+
+			if (Descriptor == null)
+				return;
+
+			RSDT = (RSDT*)HAL.GetPhysicalMemory((Pointer)Descriptor->RsdtAddress, 0xFFFF).Address;
```
Comment (tgiphil) [inline Source/Mosa.DeviceDriver/ISA/ACPI.cs:157]: I prefer not using unsafe code and using this pattern instead: 

https://github.com/mosa/MOSA-Project/blob/master/Source/Mosa.Kernel.BareMetal/MultibootSpecification/MultibootV1.cs

**Review by tgiphil — COMMENTED** [review]
See comments.

File: Source/Mosa.DeviceDriver/ISA/ACPI.cs (hunk 3 above)
Comment (AnErrupTion) [reply Source/Mosa.DeviceDriver/ISA/ACPI.cs:157]: The problem is that then the code will be huge, and the offsets of ACPI tables aren't specified anywhere on the osdev wiki, so it's close to impossible.

**Review by tgiphil — APPROVED** [review]

**Pull request merged**